### PR TITLE
[videoplayer] Teletext: refine fix 714b808

### PR DESCRIPTION
**Pull request by ace20022**
Now the last opened page gets restored.

### CODE REVIEW

**Comment by ace20022** [comment]
@Hohlmackenreiter this should fix your issue.

**Review by FernetMenta — APPROVED** [review]

File: xbmc/cores/VideoPlayer/VideoPlayerTeletext.h [hunk 1]
```diff
@@ -61,7 +61,7 @@ class CDVDTeletextData : public CThread, public IDVDStreamPlayer
   void AllocateCache(int magazine);
 
   int m_speed;
-  TextCacheStruct_t  m_TXTCache;
+  std::shared_ptr<TextCacheStruct_t> m_TXTCache = std::make_shared<TextCacheStruct_t>();
```
Comment (garbear) [inline xbmc/cores/VideoPlayer/VideoPlayerTeletext.h]: Just `std::shared_ptr<TextCacheStruct_t> m_TXTCache;` has the same effect

Comment (ace20022) [reply xbmc/cores/VideoPlayer/VideoPlayerTeletext.h]: No, then m_TXTCache is the empty shared pointer and it crashes in the c'tor.

Comment (garbear) [reply xbmc/cores/VideoPlayer/VideoPlayerTeletext.h]: Oh yea, nvm :)

**Pull request merged**